### DISCUSSION TITLE
Is there a way to do fractional scaling (fractional zoom) in Mapsforge 0.16.0 on Android?

### DISCUSSION THREAD

**Discussion by wharnisch** [mapsforge/mapsforge#1253] [Q&A] [answered]
I use Mapsforge 0.16.0 to show offline maps in an Android app.

To scale the map display, I use map.getModel().mapViewPosition.setZoomLevel(n). 'n' is an integer value, but I need to scale fractionally. So I can set zoom level to 13 or I can set zoom level to 14, but I can not set zoom level to 13.5, for example.

Also if the user tries to zoom the map, it always snaps to an integer zoom level.

Is there a way to overcome this, so is there a way to do fractional scaling (fractional zoom) in Mapsforge 0.16.0?

If there is currently no way to overcome this, I would like to submit a feature request for enabling fractional scaling, as almost all other map kits for Android offer this feature and I think this would be a very important feature for a better user experience.

**Comment by devemux86** [comment, marked as answer]
Better use [VTM OpenGL map library](https://github.com/mapsforge/vtm) (compatible with Mapsforge maps)
which has fractional scaling, map rotation / tilt and many more features.
